dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/webapp/WebApp.java

```diff
@@ -4969,6 +4982,20 @@ public void handleRequest(ServletRequest request, ServletResponse response, Http
                 }
             }
         } catch (Throwable th) {
+            //PI80786
+            if (WCCustomProperties.SET_400_SC_ON_TOO_MANY_PARENT_DIRS) { 
+                if(th.getMessage().contains("is invalid because it contains more references to parent directories")){
+                    if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
```

Comment:
Phu, is there not a message ID or something that we can look for here? I'm concerned about gating logic based on the text of an exception.

Reply:
Do you know what the call stack is that gets us from the WebContainer into the WSUtil.resolveURI(String)? I'm curious if there is any way for us to check this error condition before we call into WSUtil.resolveURI or something of that nature. Since it looks like WSUtil is used by many components changing that exception that is thrown could cause issues as well so that is probably not an option.

Reply:
There are 3 cases for the request

 http://localhost:9081/pmdinhApp%2F..%2F..%2F..%2Fboot.ini

1.  No match servlet mapping, no security
Stack Dump = java.lang.IllegalArgumentException: The specified URI, /../../../boot.ini, is invalid because it contains more references to parent directories ("..") than is possible.
	at com.ibm.ws.util.WSUtil.resolveURI(WSUtil.java:107)
	at com.ibm.ws.webcontainer.extension.DefaultExtensionProcessor.handleRequest(DefaultExtensionProcessor.java:401)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.invokeTarget(WebAppFilterChain.java:147)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:89)
	at com.ibm.wl3eng.servlets.TestFilter.doFilter(TestFilter.java:25)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:197)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:86)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.doFilter(WebAppFilterManager.java:995)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1131)
	at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:4975)

2. Matched servlet mapping (like /* or / ...usually Springs uses this type of mapping).  There will be no exception thrown by us since it will be handled by the servlet which may check and throw an exception later on.

3. App security enabled.   Sec also use same WSUtil.resolveURI() during their security preInvoke().